render/render.js

```diff
@@ -91,10 +145,15 @@ module.exports = function($window) {
 		return element
 	}
 	function createComponent(vnode, hooks, ns) {
-		vnode.state = copy(vnode.tag)
+		if (typeof vnode.tag !== "function") {
+			vnode.state = {}
```

Comment:
the above line looks like an oversight (also looking on the recycle thing, but don't hold your breath, I'm discovering the render code as I go)